### PR TITLE
added mixpanel analyticst to show pwa use

### PR DESCRIPTION
**Pull request by Israellund**
## Link to Issue
Closes: #7859 

## Description of Changes
-Added Mixpanel analytics to pwa use
**EDIT:**
-added `isPWA` to headers to pass to backend instances of Mixpanel calls
-refactored .then() call to be inside try/catch
-deleted errant console.log

## Test Plan
-comment out lines 102-112 in `main.ts` so that localhost doesn't redirect to HTTPS
-using your IP address, open localhost on your phone. ex: 192.168.1.213:8080
-install pwa on desktop
-login in or go to Home screen on pwa
-open Mixpanel events dev log and and confirm you see "PWA in use" when going to the Home screen
-uncomment the lines in `main.ts`
-rerun `pnpm start`
-open localhost on desktop
-open Mixpanel events dev log
-confirm that you see "PWA NOT in use" when going to the Home screen on desktop
-click around to different functionalities of the platform, like creating a thread, creating a comment, editing groups, creating a community, joining a community, create a topic, update a topic, pink a snapshot proposal, link a template, link a discussion, link a url, enable/disable directory page, log in, etc. 
-confirm you see the property of isPWA: true int he Mixpanel Dev Events tab (you may have to refresh a few times as it may take a minute to come through)

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Comment by jnaviask** [comment]
Have we confirmed that emitting this event is sufficient to filter user events by PWA or not? The idea is we should be able to aggregate other metrics by PWA/not, so it's possible we need something like a tag rather than an event.

Even if we intend to use an event, we should ensure the name of the event adequately describes the specific event we're capturing (e.g. PWA installed), rather than describing a general application state as it currently does ("in use" is not an event but a state).

**Comment by Israellund** [comment]
@jnaviask Yeah, that's one thing I realized yesterday. Currently have switched gears to work on bugs that are preventing community creation, but when I get back to this ticket later today, I plan to add a `isPWA: <boolean>` property to each mixpanel invocation that is based off of `isAddedToHomeScreen` that is returned from `useAppStatus()`

**Comment by jnaviask** [comment]
@Israellund sounds good. Drafting this PR until it's ready for review.

File: packages/commonwealth/client/scripts/views/components/SublayoutHeader/useJoinCommunity.tsx [hunk 1]
```diff
@@ -92,12 +95,15 @@ const useJoinCommunity = () => {
...
     activeChainId,
+    isPWA,
   }: {
     address: string;
     communityId: string;
     communityChainBase: string;
     activeChainId?: string;
+    isPWA?: boolean;
   }) => {
+    console.log('isPWA: ', isPWA);
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/SublayoutHeader/useJoinCommunity.tsx:106]: console removal

File: packages/commonwealth/client/scripts/views/components/SublayoutHeader/useJoinCommunity.tsx [hunk 2]
```diff
@@ -185,6 +191,8 @@ const useJoinCommunity = () => {
 
         const address = originAddressInfo.address;
 
+        console.log('isAddedToHomeScreen: ', isAddedToHomeScreen);
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/components/SublayoutHeader/useJoinCommunity.tsx:194]: console

File: packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/Update/UpdateCommunityGroupPage.tsx [hunk 3]
```diff
@@ -122,6 +128,8 @@ const UpdateCommunityGroupPage = ({ groupId }: { groupId: string }) => {
         }}
         onSubmit={(values) => {
           const payload = makeGroupDataBaseAPIPayload(values, allowedAddresses);
+          payload.isPWA = isAddedToHomeScreen;
+          console.log('payload: ', payload);
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/Update/UpdateCommunityGroupPage.tsx:132]: console

File: packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/common/helpers/index.ts [hunk 4]
```diff
@@ -27,6 +27,7 @@ export const makeGroupDataBaseAPIPayload = (
           formSubmitValues.requirements.length + extraRequrirements
         : formSubmitValues.requirementsToFulfill,
     requirements: [],
+    isPWA: undefined as boolean | undefined,
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/common/helpers/index.ts:30]: shouldn't we use the `isAddedToHomeScreen` logic here

File: packages/commonwealth/client/scripts/views/pages/user_dashboard/index.tsx [hunk 5]
```diff
@@ -58,6 +65,21 @@ const UserDashboard = (props: UserDashboardProps) => {
...
+            isPWA: isAddedToHomeScreen,
+          },
+        }
+      : {
+          payload: {
+            event: MixpanelPWAEvent.PWA_NOT_USED,
+            isPWA: isAddedToHomeScreen,
+          },
+        },
+  );
```
Comment (mzparacha) [inline packages/commonwealth/client/scripts/views/pages/user_dashboard/index.tsx:82]:
```diff
- useBrowserAnalyticsTrack(
-   isAddedToHomeScreen
-    ? {
-        payload: {
-          event: MixpanelPWAEvent.PWA_USED,
-          isPWA: isAddedToHomeScreen,
-        },
-      }
-    : {
-        payload: {
-          event: MixpanelPWAEvent.PWA_NOT_USED,
-          isPWA: isAddedToHomeScreen,
-        },
-      },
-);
+ useBrowserAnalyticsTrack(
+   {
+     payload: {
+       event: isAddedToHomeScreen ? MixpanelPWAEvent.PWA_USED : MixpanelPWAEvent.PWA_NOT_USED,
+       isPWA: isAddedToHomeScreen,
+     },
+   }
+);
```

File: packages/commonwealth/server/controllers/server_analytics_controller.ts [hunk 6]
```diff
@@ -19,7 +19,9 @@ export class ServerAnalyticsController {
         ...(host && { isCustomDomain: config.SERVER_URL.includes(host) }),
       };
     }
+    const reqIsPWA = req?.headers?.['ispwa'] === 'true';
```
Comment (mzparacha) [inline packages/commonwealth/server/controllers/server_analytics_controller.ts:22]: i think `req?.headers?.['ispwa']` should be `req?.headers?.['isPWA']`?

**Review by mzparacha — COMMENTED** [review]
Nice, left some feedback above.

Nit: It seems like we want to add `isPWA` header to every API request, I think a simpler case for that would be to use `axios` request interceptors where we could listen to every outgoing API request and add the `isPWA` header in a few LOC's. But again this is not a blocker.

**Comment by Israellund** [comment]
@mzparacha cleaned up the logs and made some changes based on your suggestions. Let me know what you think now.

**Review by mzparacha — APPROVED** [review]

**Review by masvelio — APPROVED** [review]

**Pull request merged**